README.md

```diff
@@ -1,36 +1,115 @@
 # silverstripe-encrypt-at-rest
 
-This module allows data to be encrypted in the database, but be decrypted when extracted from the database, using a
-secret key (hopefully) known only by the web server.
+This module allows Silverstripe CMS ORM data to be encrypted before being stored in the database, and automatically decrypted before using within your application. To do this, we use a secret key known only by the web server.
 
-*Note:* This does not provide significant protection except in the case of database compromise. It should be used as
-part of a layered security strategy. This is because the key is still available on the web server, so if remote code
-execution is achieved by an attacker, they will be able to read both the database *and* the encryption key, thereby
-decrypting the content.
 
-*Note:* This module is not yet ready for real use, it's currently v0.0.1 material.
+## Caveats to understand
+* It's important to note that this module does not guarantee the security of your data completely. You should only use this as a protection measure if you fully understand how the module operates. In most cases, encrypting the entire database is both adequate and similarly effective. Only use this module to encrypt data at-rest (on a field-by-field basis) if your layered protection strategy requires and accomodates it. To be clear - when encrypting data at rest, the data must be decrypted before being used. In almost all cases, the web server hosting the website is far more accessible to attacks than the the database server, meaning that an attacker who can compromise your web server will have access to both the database and the encryption key used to encrypt the data.
+* Encrypting and decrypting data on a field-by-field basis has a performance overhead, which may produce undesirable results in your project.
+* This module uses the `defuse/php-encryption` library under the hood, which prefers strong security over performance. Encrypting lots of fields on a `DataObject` can significantly slow down any operations that read or write large amounts of data (for example `ModelAdmin` views in the CMS that render 50+ records at once can take many seconds of processing power just to decrypt fields). Care should be taken to ensure only the minimal set of data is encrypted, and that this data does not need to be used frequently.
+
+
+## Requirements
+* SilverStripe CMS 4
+* Requires a version of silverstripe-framework that includes [this pull request](https://github.com/silverstripe/silverstripe-framework/pull/10047), or for this PR to be manually patched into your version of framework.
```

Comment:
PR reference not necessary anymore

Reply:
Good spot, thanks @Firesphere. Agree, the PR is merged so we should just target this at `^4.9` (new extension point was added in 4.9.0).